### PR TITLE
Map math classes

### PR DESCRIPTION
**Pull request by enbrain**

### CODE REVIEW

File: mappings/net/minecraft/util/math/Direction.mapping [hunk 1]
```diff
@@ -76,6 +91,16 @@ CLASS net/minecraft/class_2350 net/minecraft/util/math/Direction
...
+	METHOD method_35832 fromVector (Lnet/minecraft/class_2338;)Lnet/minecraft/class_2350;
+		ARG 0 pos
+	METHOD method_35833 rotateClockwise (Lnet/minecraft/class_2350$class_2351;)Lnet/minecraft/class_2350;
+		ARG 1 axis
+	METHOD method_35834 rotateCounterclockwise (Lnet/minecraft/class_2350$class_2351;)Lnet/minecraft/class_2350;
+		ARG 1 axis
+	METHOD method_35835 rotateXClockwise ()Lnet/minecraft/class_2350;
+	METHOD method_35836 rotateXCounterclockwise ()Lnet/minecraft/class_2350;
+	METHOD method_35837 rotateZClockwise ()Lnet/minecraft/class_2350;
+	METHOD method_35838 rotateZCoutnerclockwise ()Lnet/minecraft/class_2350;
```
Comment (YanisBft) [inline mappings/net/minecraft/util/math/Direction.mapping]:
```suggestion
	METHOD method_35838 rotateZCounterclockwise ()Lnet/minecraft/class_2350;
```

**Review by YanisBft — COMMENTED** [review]
Btw, why is  `Rect2i` not in client/util/math?

**Review by YanisBft — APPROVED** [review]

File: mappings/net/minecraft/util/math/EulerAngle.mapping [hunk 2]
```diff
@@ -14,3 +14,6 @@ CLASS net/minecraft/class_2379 net/minecraft/util/math/EulerAngle
 	METHOD method_10256 getPitch ()F
 	METHOD method_10257 getYaw ()F
 	METHOD method_10258 getRoll ()F
+	METHOD method_35845 getWrappedPitch ()F
+	METHOD method_35846 getWrappedYaw ()F
+	METHOD method_35847 getWrappedRoll ()F
```
Comment (haykam821) [inline mappings/net/minecraft/util/math/EulerAngle.mapping]: Document the range of the return value. For example, would it be `[-180, 180)` or `[0, 360)`?

**Review by haykam821 — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/util/math/MathHelper.mapping [hunk 3]
```diff
@@ -151,8 +152,10 @@ CLASS net/minecraft/class_3532 net/minecraft/util/math/MathHelper
 		ARG 0 a
 		ARG 2 b
 	METHOD method_15392 wrapDegrees (I)I
+		COMMENT Wraps an angle in degrees to the interval [-180, 180).
 		ARG 0 degrees
 	METHOD method_15393 wrapDegrees (F)F
+		COMMENT Wraps an angle in degrees to the interval [-180, 180).
```
Comment (liach) [inline mappings/net/minecraft/util/math/MathHelper.mapping]:
```suggestion
		COMMENT Wraps an angle in degrees to the interval {@code [-180, 180)}.
```

File: mappings/net/minecraft/util/math/MathHelper.mapping [hunk 4]
```diff
@@ -151,8 +152,10 @@ CLASS net/minecraft/class_3532 net/minecraft/util/math/MathHelper
 		ARG 0 a
 		ARG 2 b
 	METHOD method_15392 wrapDegrees (I)I
+		COMMENT Wraps an angle in degrees to the interval [-180, 180).
```
Comment (liach) [inline mappings/net/minecraft/util/math/MathHelper.mapping]:
```suggestion
		COMMENT Wraps an angle in degrees to the interval {@code [-180, 180)}.
```

File: mappings/net/minecraft/util/math/MathHelper.mapping [hunk 5]
```diff
@@ -7,6 +7,7 @@ CLASS net/minecraft/class_3532 net/minecraft/util/math/MathHelper
 	FIELD field_15727 ARCSINE_TABLE [D
 	FIELD field_15728 SMALLEST_FRACTION_FREE_DOUBLE D
 	METHOD method_15338 wrapDegrees (D)D
+		COMMENT Wraps an angle in degrees to the interval [-180, 180).
```
Comment (liach) [inline mappings/net/minecraft/util/math/MathHelper.mapping]:
```suggestion
		COMMENT Wraps an angle in degrees to the interval {@code [-180, 180)}.
```

File: mappings/net/minecraft/util/math/EulerAngle.mapping [hunk 6]
```diff
@@ -14,3 +14,9 @@ CLASS net/minecraft/class_2379 net/minecraft/util/math/EulerAngle
 	METHOD method_10256 getPitch ()F
 	METHOD method_10257 getYaw ()F
 	METHOD method_10258 getRoll ()F
+	METHOD method_35845 getWrappedPitch ()F
+		COMMENT Returns the pitch that is wrapped to the interval [-180, 180).
+	METHOD method_35846 getWrappedYaw ()F
+		COMMENT Returns the yaw that is wrapped to the interval [-180, 180).
+	METHOD method_35847 getWrappedRoll ()F
+		COMMENT Returns the roll that is wrapped to the interval [-180, 180).
```
Comment (liach) [inline mappings/net/minecraft/util/math/EulerAngle.mapping]:
```suggestion
		COMMENT Returns the roll that is wrapped to the interval {@code [-180, 180)}.
```

File: mappings/net/minecraft/util/math/EulerAngle.mapping [hunk 7]
```diff
@@ -14,3 +14,9 @@ CLASS net/minecraft/class_2379 net/minecraft/util/math/EulerAngle
 	METHOD method_10256 getPitch ()F
 	METHOD method_10257 getYaw ()F
 	METHOD method_10258 getRoll ()F
+	METHOD method_35845 getWrappedPitch ()F
+		COMMENT Returns the pitch that is wrapped to the interval [-180, 180).
+	METHOD method_35846 getWrappedYaw ()F
+		COMMENT Returns the yaw that is wrapped to the interval [-180, 180).
```
Comment (liach) [inline mappings/net/minecraft/util/math/EulerAngle.mapping]:
```suggestion
		COMMENT Returns the yaw that is wrapped to the interval {@code [-180, 180)}.
```

File: mappings/net/minecraft/util/math/EulerAngle.mapping [hunk 8]
```diff
@@ -14,3 +14,9 @@ CLASS net/minecraft/class_2379 net/minecraft/util/math/EulerAngle
 	METHOD method_10256 getPitch ()F
 	METHOD method_10257 getYaw ()F
 	METHOD method_10258 getRoll ()F
+	METHOD method_35845 getWrappedPitch ()F
+		COMMENT Returns the pitch that is wrapped to the interval [-180, 180).
```
Comment (liach) [inline mappings/net/minecraft/util/math/EulerAngle.mapping]:
```suggestion
		COMMENT Returns the pitch that is wrapped to the interval {@code [-180, 180)}.
```

**Review by liach — APPROVED** [review]
Stylistic stuff for javadoc

**Comment by liach** [comment]
@enbrain mind fix your merge conflict real quick?

**Comment by Player3324** [comment]
multiplyModel is incorrect: MC uses a projection and a modelview matrix, both the "modelMatrix" and the normalMatrix affect the combined ModelView transformation step (local space to camera space). IMO all instances of "model(Matrix)" in MatrixStack should be replaced with position(Matrix) or similar - the matrix used to transform positions.

**Comment by enbrain** [comment]
@sfPlayer1 I'm not sure if your suggestion should be applied, so I will just revert `multiplyModel` for now. I opened an issue about model matrix stuff #2377.

Thanks for review!

**Comment by Player3324** [comment]
Is there a reason why this one is not yet marked ready to merge?

**Comment by enbrain** [comment]
Apparently liach wants an approval from player but imo this should be ready to merge as it got 2 approvals already.

**Comment by liach** [comment]
> 
> 
> Apparently liach wants an approval from player but imo this should be ready to merge as it got 2 approvals already.

Yes. I usually expect an approval after adjusting for request. Will take note on this.

File: mappings/net/minecraft/util/math/MathHelper.mapping [hunk 9]
```diff
@@ -277,5 +280,16 @@ CLASS net/minecraft/class_3532 net/minecraft/util/math/MathHelper
...
 		ARG 1 y
 		ARG 3 z
+	METHOD method_34939 clamp (BBB)B
+		ARG 0 value
+		ARG 1 min
+		ARG 2 max
+	METHOD method_34947 parseDouble (Ljava/lang/String;D)D
+		ARG 0 string
+		ARG 1 fallback
+	METHOD method_34951 cumulativeProportions ([D)[D
```
Comment (Juuxel) [inline mappings/net/minecraft/util/math/MathHelper.mapping]: What does this method do?

Comment (enbrain) [reply mappings/net/minecraft/util/math/MathHelper.mapping]: This method calculates `(values[0] + ... + values[i]) / (values[0] + ... + values[values.length - 1])` for each `i`.

Comment (liach) [reply mappings/net/minecraft/util/math/MathHelper.mapping]: Then I suggest
```suggestion
	METHOD method_34951 getCumulativeDistribution ([D)[D
```
As this calculates the [cumulative distribution function](https://en.wikipedia.org/wiki/Cumulative_distribution_function) from the given array as a discrete probabilities.

However, if this method has usages, then it's better to decide from the usage of the method than from what this method does.

**Pull request merged**